interface/patient_file/pos_checkout_normal.php

```diff
@@ -580,7 +580,7 @@ function markTaxes($taxrates)
             if ($GLOBALS['discount_by_money']) {
                 $amount  = sprintf('%01.2f', trim($_POST['form_discount']));
             } else {
-                $amount  = sprintf('%01.2f', trim($_POST['form_discount']) * $form_amount / 100);
+                $amount  = sprintf('%01.2f', trim($_POST['form_discount']) * $form_amount/(100-(trim($_POST['form_discount']))));
```

Comment:
I thought the bug was that the discount needs to be on the total amount instead of the form amount.  I'm not seeing how this fixes the original problem as the form_amount is still being used here.  Also there is a potential division by zero error here if form_discount is set to 100 so you'll need to check and make sure that never happens.

@stephenwaite You probably know this file better than I do, is there a better variable that @surfacehazard should use here instead of form_amount (after he's fixed the divide by zero bug).

Reply:
The bug is that the discount is being REapplied to the form_amount on this line. The updated calculation gets you back to the total amount before applying the discount. I can fix the division by zero issue for this code, but it is already broken somewhere else in the file (i.e. if you give a 100% discount, the receipt will show the whole balance outstanding).

Reply:
After typing that, thought occurred to me that perhaps no calculation should happen here at all.  `$amount = sprintf('%01.2f', $form_amount);` might do the same thing.

Reply:
@adunsulag total amount = discounted amount / (100 - discount percentage) * 100

Reply:
not that familiar with it actually since it interferes with insurance billing if it's used for copays as @sunsetsystems notes
```
(4) On checkout, the billing and drug_sales table entries are marked
 *     as billed and so become unavailable for further billing.
```